pkgs/development/python-modules/txtai/default.nix

```diff
@@ -176,23 +190,49 @@ buildPythonPackage {
 
   optional-dependencies = optional-dependencies;
 
-  pythonImportsCheck = [ "txtai" ];
-
-  # some tests hang forever
-  doCheck = false;
-
-  preCheck = ''
-    export TRANSFORMERS_CACHE=$(mktemp -d)
+  # The Python imports check runs huggingface-hub which needs a writable directory
+  postInstall = ''
+    export HF_HOME=$(mktemp -d)
   '';
 
+  pythonImportsCheck = [ "txtai" ];
+
   nativeCheckInputs = [
-    unittestCheckHook
-  ] ++ optional-dependencies.api ++ optional-dependencies.similarity;
+    pytestCheckHook
+  ] ++ optional-dependencies.ann ++ optional-dependencies.api ++ optional-dependencies.similarity;
+
+  checkInputs = [
+    httpx
+    msgpack
+    python-multipart
+    sqlalchemy
+  ];
+
+  # The deselected paths depend on the huggingface hub and should be run as a passthru test
+  pytestFlagsArray = [
+    "test/python/test*.py"
+    "--deselect=test/python/testcloud.py"
+    "--deselect=test/python/testconsole.py"
+    "--deselect=test/python/testembeddings.py"
+    "--deselect=test/python/testgraph.py"
+    "--deselect=test/python/testapi/testembeddings.py"
+    "--deselect=test/python/testapi/testpipelines.py"
+    "--deselect=test/python/testapi/testworkflow.py"
+    "--deselect=test/python/testdatabase/testclient.py"
+    "--deselect=test/python/testdatabase/testduckdb.py"
+    "--deselect=test/python/testdatabase/testencoder.py"
+    "--deselect=test/python/testworkflow.py"
+  ];
```

Comment:
Can't we use `disabledTests` for those ?

Reply:
Sadly, no. The failures happen when initializing the classes holding the tests (no way to list those in disabledTests).  Also, `disabledTestPaths` didn't work. I put them here after two hours of trying with disabledTestPaths.